vault/mount.go

```diff
@@ -40,6 +40,14 @@ const (
 	mountTableType = "mounts"
 )
 
+type ListingVisiblityType string
+
+const (
+	ListingVisibilityHidden ListingVisiblityType = ""
+	ListingVisibilityAuth   ListingVisiblityType = "auth"
```

Comment:
If we are not going ahead with string type for `ListingVisibility`, I assume that you intend to use these variables all the way, which is only not done yet, I assume. Just a reminder to do so :-)